src/Symfony/Component/Console/Output/ConsoleSectionOutput.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Output;
+
+use Symfony\Component\Console\Formatter\OutputFormatterInterface;
+use Symfony\Component\Console\Helper\Helper;
+use Symfony\Component\Console\Terminal;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ * @author Gabriel Ostrolucký <gabriel.ostrolucky@gmail.com>
+ */
+class ConsoleSectionOutput extends StreamOutput
```

Comment:
does it makes sense to use this class totally without `ConsoleOutput` ?
especially when having the connection between them via `&$sections`

Reply:
It's fine to construct this class completely without `ConsoleOutput`. Coupling it to `ConsoleOutput` more strictly does not make sense

Reply:
I don't think this class should ever be used outside of ConsoleOutput, as it is an implementation detail of the console. I agree this class can be marked as internal

Reply:
Well, using this feature as an end user involves to use the public api of this class, marking it as internal without providing an interface would be weird. I'm not sure we want to introduce an interface for that, so letting it as is (maybe `@final`) with some documentation is probably fine.

Reply:
As I have shown in different comment, there are situations when its a must to manipulate with order of sections manually, which can be done only when having access to array container. section() method provides only the most basic operation (creating new section at the bottom of the screen)

Reply:
That's the thing. manually creating a section without having it anchored to full output class is weird and imo shall not be allowed.
Currently Output is aware about sections and sections one about another. so we have big part which could contain small parts.
Even if it technically possible to create standalone section instance, what is the usage of that?

Thus section concrete class shall IMO be final, created only by output class itself, so interface to provide info how to use it.

Reply:
I don't see anything weird about that. ConsoleOutput currently serves only as a simple stupid helper for creating new output sections at the bottom of the screen. I'm out of luck if I need to create another new section at the top of existing sections. Anyway I could currently do this even if that class is final, because as I said ConsoleOutput serves as a stupid helper only and I can bypass it, final or not. So I don't see what does it solve. 

I would need to ask symfony community to stop making final classes though, you cannot anticipate what will user space need to do. `@final` annotation is enough to mark extensions as not covered by BC policy. I wouldn't even do that though, this class has lean API and high potential to be further extended.

Marking it `@internal` is out of the question as @chalasr explained.

Reply:
I disagree about all `final`-about statements. But I guess I'm just defensive.

No, he didn't ;) he said it would be weird when done `without providing an interface`

If you construct section out of the box without `ConsoleOutput` helping here, you need to manually interact with other sections or have that `&$sections`. Yet, it's private in both, `SectionOutput` and `ConsoleOutput`. It will be way cleaner if all will be done via a helper, if you need to inject a section (put it not at the end) after some of them were already created, adjust the helper to allows that.
I'm big critic of passing reference around different instances and manually managing them, so if we need to have them, let us make it internal implementation detail, not sth that user need to deal if he try to not use helper.